ext/node/polyfill.rs

```diff
@@ -1,229 +1,209 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 
-use deno_core::error::generic_error;
-use deno_core::error::AnyError;
-use deno_core::url::Url;
-use deno_core::ModuleSpecifier;
-
-// TODO(bartlomieju): seems super wasteful to parse the specifier each time
-pub fn resolve_builtin_node_module(module_name: &str) -> Result<Url, AnyError> {
-  if let Some(module) = find_builtin_node_module(module_name) {
-    return Ok(ModuleSpecifier::parse(module.specifier).unwrap());
-  }
-
-  Err(generic_error(format!(
-    "Unknown built-in \"node:\" module: {module_name}"
-  )))
-}
-
-fn find_builtin_node_module(module_name: &str) -> Option<&NodeModulePolyfill> {
+pub fn is_builtin_node_module(module_name: &str) -> bool {
   SUPPORTED_BUILTIN_NODE_MODULES
     .iter()
-    .find(|m| m.name == module_name)
-}
-
-pub fn is_builtin_node_module(module_name: &str) -> bool {
-  find_builtin_node_module(module_name).is_some()
+    .any(|m| m.specifier.strip_prefix("node:").unwrap() == module_name)
```

Comment:
So this implies that `module_name` passed to this function doesn't have `node:` prefix - Could you add a docstring about it to this function?